### DISCUSSION TITLE
Dynamically load new data in ActiveAdmin

### DISCUSSION THREAD

**Discussion by moptonal** [activeadmin/activeadmin#8391] [Ideas]
I'm using activeadmin daily and there's a tab I consistently have open that I go back to and manually refresh. Would be great if there was a way to automatically load new data dynamically.

**Comment by jaynetics** [comment]
you can add JS in partials, but also in arbre, e.g.

```ruby
index do
  script { 'setTimeout(() => window.location.reload(), 5 * 60 * 1000)'.html_safe }
end
```

**Comment by matteeyah** [comment]
Would be even better if it could dynamically load the new data.

**Reply by moptonal** [reply]
Updating the issue to reflect this because that's a better description of what I'm looking for.

**Comment by jsntv200** [comment]
I've been playing around with an activeadmin v4 setup and had no issue getting Turbo 8 installed and updating an index page. Tested with `Redis` and the no redis option [actioncable-enhanced-postgresql-adapter](https://github.com/reclaim-the-stack/actioncable-enhanced-postgresql-adapter) without issue.

Loads of options for updating content with Turbo but simplest is to trigger a page reload with `broadcast_refresh_later_to` or `broadcast_refresh_to`.

```ruby
# app/model/post.rb
class Post < ApplicationRecord
  after_commit -> { broadcast_refresh_later_to :posts }
end
```

```ruby
# app/admin/posts.rb
ActiveAdmin.register Post do
  index do
    column :title
    column :body
    actions
    div { turbo_stream_from :posts }
  end
end
```

You can prevent page jumps by adding the following meta tag, to load the required js I needed a custom `_html_head.html.erb`, so added it there:
```ruby
# app/views/active_admin/_html_head.html.erb
<meta name="turbo-refresh-scroll" content="preserve">
```
